ast35/Python/ast.c

```diff
@@ -1304,10 +1320,16 @@ handle_keywordonly_args(struct compiling *c, const node *n, int start,
                 if (!arg)
                     goto error;
                 asdl_seq_SET(kwonlyargs, j++, arg);
-                i += 2; /* the name and the comma */
+                i += 1 + (TYPE(CHILD(n, i + 1)) == COMMA); /* the name and the comma, if present */
                 break;
             case DOUBLESTAR:
                 return i;
+            case TYPE_COMMENT:
```

Comment:
I'd stick this case directly under the `vfpdef`/`tfpdef` case -- it's sort of an optional adjunct to it, and its logic is in fact (as the comment indicates) quite intimately tied to the other. The `DOUBLESTAR` case is where we break out of this loop to effectively jump to a new state in the (implicit) state machine in the parent loop, so it makes sense to have it down by the bottom.